automated install/basic-install.sh

```diff
@@ -1023,106 +1170,122 @@ setDNS() {
     esac
 }
 
 # Allow the user to set their FTL privacy level
 setPrivacyLevel() {
-    local LevelCommand
-    local LevelOptions
-
-    LevelCommand=(whiptail --separate-output --radiolist "Select a privacy mode for FTL. https://docs.pi-hole.net/ftldns/privacylevels/" "${r}" "${c}" 6)
-
     # The default selection is level 0
-    LevelOptions=(
-        "0" "Show everything" on
-        "1" "Hide domains" off
-        "2" "Hide domains and clients" off
-        "3" "Anonymous mode" off
-    )
-
-    # Get the user's choice
-    PRIVACY_LEVEL=$("${LevelCommand[@]}" "${LevelOptions[@]}" 2>&1 >/dev/tty) || (echo -e "  ${COL_LIGHT_RED}Cancel was selected, exiting installer${COL_NC}" && exit 1)
-
-    printf "  %b Privacy level %d" "${INFO}" "${PRIVACY_LEVEL}"
+    PRIVACY_LEVEL=$(dialog --no-shadow --clear --output-fd 1 \
+        --radiolist "Select a privacy mode for FTL. https://docs.pi-hole.net/ftldns/privacylevels/" \
+        "${r}" "${c}" 6 \
+        "0" "Show everything" on \
+        "1" "Hide domains" off \
+        "2" "Hide domains and clients" off \
+        "3" "Anonymous mode" off)
+
+        result=$?
+        case ${result} in
+            "${DIALOG_OK}")
+                printf "  %b Using privacy level: %s\\n" "${INFO}" "${PRIVACY_LEVEL}"
+                ;;
+            "${DIALOG_CANCEL}" | "${DIALOG_ESC}")
+                printf "  %b Cancelled privacy level selection.%b\\n" "${COL_LIGHT_RED}" "${COL_NC}"
+                exit 1
+                ;;
+        esac
 }
 
 # Function to ask the user if they want to install the dashboard
 setAdminFlag() {
-    # Local, named variables
-    local WebToggleCommand
-    local WebChooseOptions
-    local WebChoices
-
     # Similar to the logging function, ask what the user wants
-    WebToggleCommand=(whiptail --separate-output --radiolist "Do you wish to install the web admin interface?" "${r}" "${c}" 6)
-    # with the default being enabled
-    WebChooseOptions=("On (Recommended)" "" on
-        Off "" off)
-    WebChoices=$("${WebToggleCommand[@]}" "${WebChooseOptions[@]}" 2>&1 >/dev/tty) || (printf "  %bCancel was selected, exiting installer%b\\n" "${COL_LIGHT_RED}" "${COL_NC}" && exit 1)
-    # Depending on their choice
-    case ${WebChoices} in
-        "On (Recommended)")
-            printf "  %b Web Interface On\\n" "${INFO}"
-            # Set it to true
+    dialog --no-shadow --clear \
+        --backtitle "Pihole Installation" \
+        --title "Admin Web Interface" \
+        --yesno "\\n\\nDo you want to install the Admin Web Interface?" \
+        "${r}" "${c}"
+
+    result=$?
+    case ${result} in
+        "${DIALOG_OK}")
+            # If they chose yes,
+            printf " %b Installing Admin Web Interface\\n" "${INFO}"
+            # Set the flag to install the web interface
             INSTALL_WEB_INTERFACE=true
             ;;
-        Off)
-            printf "  %b Web Interface Off\\n" "${INFO}"
-            # or false
+        "${DIALOG_CANCEL}")
```

Comment:
Selecting `No` also exits the installer.

Reply:
Should be fixed. That was caused by `set -e` and not having a guard when dialog returns non-zero for No/Cancel/Escape.